mailchimp_subscriber.py

```diff
@@ -15,21 +61,39 @@ def load_conf(conf_file):
 
 def load_users(users_file):
     """Read the email addresses from disk.
-    Returns a set of email addresses"""
+    Returns a set of Client objects"""
+    clients = dict()
     with open(users_file, 'r') as f:
-        return {l.rstrip() for l in f.readlines()}
+        for row in csv.reader(f):
+            # Note that this isn't testing for multiple appearances of the same
+            # client. If theres more than one, it takes the last appearence.
+            try:
+                clients[row[0]] = Client(row)
+            except ValueError:
+                pass
+
+    return clients
 
 
 def process_users(users, list_id, mc_user, mc_key):
     """Takes a list of emails and returns a list of those not subscribed
     to the MailChimp list"""
-    client = MailChimp(mc_user, mc_key)
-    contacts = client.lists.members.all(list_id, get_all=True,
-                                        status='subscribed',
-                                        fields="members.email_address")
-    email_addresses = {contact['email_address']
-                       for contact in contacts['members']}
-    return users.difference(email_addresses)
+    mc_client = MailChimp(mc_user, mc_key)
+    for client in users.values:
+        set_mailchimp_status(client, mc_client, list_id)
+    for client in users.values:
+        add_user(client, mc_client, list_id)
+
+
+def add_user(client, mc_client, list_id):
+    if (client.mailchimp_status == 'pending' or
```

Comment:
Should this conditional include `== 'pending'`?

Reply:
Yes, I think so.  A client could have one of six different statuses.  Only in these two cases do I want it to act on them.  Is there a better way to test for these two?

Reply:
Asking as I think the code here tries to update the status to "pending" from "not present". If the user's status is already "pending", then this operation seems redundant.